src/sqlancer/Main.java

```diff
@@ -254,30 +254,36 @@ private void printState(FileWriter writer, StateToReproduce state) {
             this.globalState = globalState;
         }
 
-        public boolean execute(Query q) throws SQLException {
-            globalState.getState().logStatement(q);
-            boolean success = q.execute(globalState);
-            Main.nrSuccessfulActions.addAndGet(1);
-            return success;
-        }
-
-        public ResultSet executeAndGet(Query q) throws SQLException {
-            globalState.getState().logStatement(q);
-            ResultSet result = q.executeAndGet(globalState);
-            Main.nrSuccessfulActions.addAndGet(1);
-            return result;
-        }
-
-        public boolean fillAndExecute(Query q, String template, List<String> fills) throws SQLException {
+        public boolean execute(Query q, String... templateAndFills) throws SQLException {
             globalState.getState().logStatement(q);
-            boolean success = q.fillAndExecute(globalState, template, fills);
+            boolean success;
+            if (templateAndFills.length < 2) {
+                success = q.execute(globalState);
+            } else {
+                String template = templateAndFills[0];
+                List<String> fills = new ArrayList<>();
+                for (int i = 1; i < templateAndFills.length; i++) {
+                    fills.add(templateAndFills[i]);
+                }
+                success = q.fillAndExecute(globalState, template, fills);
```

Comment:
Is this what you had in mind? I couldn't remove the fillAndExecute method from all classes, since it prepares the SQL statements in a different way than the normal queries. That's why, I turned the execution methods in the GlobalState and Main classes into a generic mode, but there is a logic inside the execution methods in the Main class that calls different methods from the Query class.